fvtest/tril/tril/ilgen.cpp

```diff
@@ -248,6 +248,48 @@ TR::Node* Tril::TRLangBuilder::toTRNode(const ASTNode* const tree) {
          node = TR::Node::create(opcode.getOpCodeValue(), childCount);
          node->setBranchDestination(targetEntry);
      }
+     else if (opcode.isCall()) { 
+        auto compilation = TR::comp();
+
+        const auto addressArg = tree->getArgByName("address");
+        if (addressArg == nullptr) { 
+           TraceIL("  Found call without required address associated\n");
+           return nullptr; 
+        }
+
+        /* I don't want to extend the ASTValue type system to include pointers at this moment,
+         * so for now, we do the reinterpret_cast to pointer type from long
+         */
+        const auto targetAddress = reinterpret_cast<void*>(addressArg->getValue()->get<long>()); 
+
+        /* To generate a call, will create a ResolvedMethodSymbol, but we need to know the 
+         * signature. The return type is intuitable from the call node, but the arguments 
+         * must be provided explicitly, hence the args list, that mimics the args list of 
+         * (method ...) 
+         */
+        const auto argList = parseArgTypes(tree); 
+
+        auto argIlTypes = std::vector<TR::IlType*>{argList.size()};
+        auto output = argIlTypes.begin(); 
+        for (auto iter = argList.cbegin(); iter != argList.cend(); iter++) { 
+           *output = _types.PrimitiveType(*iter); 
+        }
```

Comment:
Ideally, I would prefer using `std::transform` for this, but I'll let this one pass as it requires defining an additional function (and we can't use lambdas).

Reply:
https://github.com/eclipse/omr/issues/1939